magento/pwa-studio
# Checkout Page Payment Information

## Description

Guest Payment Workflow for the new Venia Checkout workflow.

As a guest, I should be able to add a credit card in the checkout page and use it for the checkout process.

Without a payment method selected, the review order button will be disabled. Once the user clicks the review order button, the braintree drop-in will make a request for the payment nonce and save it in the apollo cache.

If a payment nonce is present in the cache, it is an indication of the payment information step being complete. Hence the payment information summary should be shown instead.

The payment information summary component should have an edit option to edit the original payment method's data. Note: For security purposes, credit card details will not be stored. They should be entered again to complete the payment edit workflow.

This component saves some stuff in the cache and some on the remote GQL server. It stores payment details like type of card and last 4 digits in the cache but the nonce in the remote server due to incompatibilities between what GQL supports and what UX wants.

Sample Credit Cards for testing: https://developers.braintreepayments.com/guides/credit-cards/testing-go-live/ruby

Deployed version: http://pr-2320.pwa-venia.com

## Related Issue
Closes [PWA-183](https://jira.corp.magento.com/browse/PWA-183)

### Verification Stakeholders
@soumya-ashok 
@schensley 
@jimbo 

### Verification Steps
1. Go to the Checkout page.
2. Complete the first 2 steps.
3. The payment information section should show up.

## Screenshots / Screen Captures (if appropriate)

![image](https://user-images.githubusercontent.com/35203638/78696535-30276500-78c5-11ea-90c8-5afb2ac319e3.png)

![image](https://user-images.githubusercontent.com/35203638/78696579-3e758100-78c5-11ea-805b-ee656310eb1f.png)

![image](https://user-images.githubusercontent.com/35203638/78696684-606f0380-78c5-11ea-8260-d2970c2a8f21.png)

## Checklist
1. Once order confirmation PR is merged, we will have to make sure that it is deleting the payment nonce and billing address data from the cache that this component is adding.
